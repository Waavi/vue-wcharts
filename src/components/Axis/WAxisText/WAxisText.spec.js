import { shallowMount } from '@vue/test-utils'
import WAxisText from './WAxisText.vue'

describe('Common/WAxisText', () => {
    const propsData = {
        x: 30,
        y: 135,
        index: 1,
        value: 'Page 1',
        styles: {
            stroke: '#444',
            fill: '#ddd',
            fontSize: 12,
        },
    }

    const provide = {
        Chart: {
            active: {
                el: null,
                event: null,
            },
            activeElements: [],
            setActive: () => undefined,
            cleanActive: () => undefined,
        },
    }

    const defaultConfig = {
        propsData,
        provide,
    }

    it(`Should be render correctly`, () => {
        const wrapper = shallowMount(WAxisText, defaultConfig)
        expect(wrapper.html()).toMatchSnapshot()
    })
})
