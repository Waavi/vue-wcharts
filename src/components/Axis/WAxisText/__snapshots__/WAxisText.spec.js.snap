// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common/WAxisText Should be render correctly 1`] = `
<text x="30" y="135" dy="" stroke="none" fill="none" font-size="14">
  Page 1
</text>
`;

exports[`Common/WAxisText Should be render correctly with custom styles 1`] = `
<text x="30" y="135" dy="" stroke="#444" fill="#ddd" font-size="12">
  Page 1
</text>
`;
